lib/imgapi.js

```diff
@@ -278,8 +278,8 @@ function IMGAPI(options) {
     } else {
         this.url = options.url;
     }
-    // _basePath: the URL subpath *without* a trailing '/'
-    this._basePath = parsed.pathname;
+    // the URL subpath *without* a trailing '/' or query params
+    this._basePath = parsed.path.split('?')[0];
```

Comment:
Why this change?  My understanding is that `parsed.pathname` *is* a field without the query params:

```
> parsed = url.parse('https://images.joyent.com/foo/?channel=asdf')
Url {
  protocol: 'https:',
  slashes: true,
  auth: null,
  host: 'images.joyent.com',
  port: null,
  hostname: 'images.joyent.com',
  hash: null,
  search: '?channel=asdf',
  query: 'channel=asdf',
  pathname: '/foo/',
  path: '/foo/?channel=asdf',
  href: 'https://images.joyent.com/foo/?channel=asdf' }
```

The current master lib/imgapi.js doesn't have this change.

Reply:
Yep, I went back and tested imgadm on SmartOS without this, and can't see why I had it originally. Reverted it now.